### ISSUE TITLE
Player cannot connect correctly to a server with resource pack. All other servers are no problem.

### ISSUE THREAD

**Issue by SonnenPingu** [GeyserMC/Geyser#4519]
### Describe the bug

He receives the error message; No connection to the world.
Then he tries again and gets a crash message from Minecraft and on the next attempt he lands directly on the server without landing on the lobby as intended.

### To Reproduce

Login to the proxy with the PS4/5.
Switch from the lobby server to the resource pack server.
See if it worked or not.
In 80% of all logins it works but 20% not.

### Expected behaviour

That the change works without "errors".

### Server Version and Plugins

Server Pandoria: [16:06:33 INFO]: This server is running Paper version git-Paper-459 (MC: 1.20.4) (Implementing API version 1.20.4-R0.1-SNAPSHOT) (Git: 88419b2)
You are 2 version(s) behind
Download the new version at: https://papermc.io/downloads/paper
Previous version: git-Paper-454 (MC: 1.20.4)
Proxy: [16:09:07 INFO]: Velocity 3.3.0-SNAPSHOT (git-75d6dcc1-b371)
[16:09:07 INFO]: Copyright 2018-2023 Velocity Contributors. Velocity is licensed under the terms of the GNU General Public License v3.
[16:09:07 INFO]: velocitypowered.com - GitHub

### Geyser Dump

https://dump.geysermc.org/o3LE4bVKdvzNEBynv9hhXLHBkawMtuM5

### Geyser Version

2.2.2-SNAPSHOT (git-master-c64e8af)",

### Minecraft: Bedrock Edition Device/Version

Playstion 4/5 

### Additional Context

https://mclo.gs/4YonNCx 
https://mclo.gs/6EtUB52 
This is also an interesting phenomenon with this first it is then it changes and is completely gone. The console hangs up completely and then it logs in correctly.

**Comment by onebeastchris** [comment]
https://mclo.gs/4YonNCx#L342
That's the interesting error which is likely causing the issues - are you always seeing this error in those 20% failed connections?

**Comment by SonnenPingu** [comment]
Yes, this error always appears exactly when it gets No connection to world again.

**Comment by onebeastchris** [comment]
This should've been resolved a while back; gonna close this. Thanks for reporting!
